src/m365/teams/commands/team/team-add.ts

```diff
@@ -187,6 +320,48 @@ class TeamsTeamAddCommand extends GraphCommand {
 
     return group!;
   }
+
+  private async retrieveMembersToAdd(members: TeamMember[], role: string, emails?: string, ids?: string, userNames?: string): Promise<any[]> {
+    let itemsToProcess: string[] = [];
+
+    if (emails) {
+      for (const email of emails.split(',')) {
+        const userId = await entraUser.getUserIdByEmail(email);
+        itemsToProcess.push(userId);
+      }
+    }
+    else if (ids) {
+      itemsToProcess = ids.split(',');
+    }
+    else if (userNames) {
+      for (const un of userNames.split(',')) {
+        const userId = await entraUser.getUserIdByUpn(un);
+        itemsToProcess.push(userId);
+      }
+    }
+
+    itemsToProcess.map((item: string) => {
+      if (!members.some((y: TeamMember) => y['user@odata.bind'] === `https://graph.microsoft.com/v1.0/users('${item}')`)) {
+        members.push({
+          '@odata.type': '#microsoft.graph.aadUserConversationMember',
+          'user@odata.bind': `https://graph.microsoft.com/v1.0/users('${item}')`,
+          roles: [role]
+        });
+      }
+      else {
+        const returnObject = members.find((y: TeamMember) => y['user@odata.bind'] === `https://graph.microsoft.com/v1.0/users('${item}')`);
```

Comment:
Let's refactor this code so that we're not doing the same lookup twice: here and in line 344